files/acl.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+const FILE_IS_PUBLIC = 'FILE_IS_PUBLIC';
+const FILE_IS_PRIVATE_AND_ALLOWED = 'FILE_IS_PRIVATE_AND_ALLOWED';
+const FILE_IS_PRIVATE_AND_DENIED = 'FILE_IS_PRIVATE_AND_DENIED';
+const FILE_NOT_FOUND = 'FILE_NOT_FOUND';
+
+function send_visibility_headers( $file_visibility, $file_path ) {
+	// Default to throwing an error so we can catch unexpected problems more easily.
+	$status_code = 500;
+	$header = false;
+
+	switch ( $file_visibility ) {
+		case FILE_IS_PUBLIC:
+			$status_code = 202;
+			break;
+
+		case FILE_IS_PRIVATE_AND_ALLOWED:
+			$status_code = 202;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_IS_PRIVATE_AND_DENIED:
+			$status_code = 403;
+			$header = 'X-Private: true';
+			break;
+
+		case FILE_NOT_FOUND:
+			// This is a signal for the server to pass the request through to the Files Service.
+			// Even though the attachment was not found within WordPress, it may still be a physical file in the Files Service.
+			$status_code = 202;
+			break;
+
+		default:
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			trigger_error( sprintf( 'Invalid file visibility (%s) ACL set for %s', $file_visibility, $file_path ), E_USER_WARNING );
+			break;
+	}
+
+	http_response_code( $status_code );
+
+	if ( $header ) {
+		header( $header );
+	}
+}
+
+/**
+ * Variant of core's attachment_url_to_postid() function
+ *
+ * @param $path (string) The path to resolve.
+ *
+ * @return (int) The found attachment ID, or 0 on failure.
+ */
+function get_file_path_from_attachment_id( $path ) {
+	global $wpdb;
+
+	$cache_key = 'path_' . md5( $path );
+	$attachment_id = wp_cache_get( $cache_key, 'files-acl' );
+	if ( false !== $attachment_id ) {
+		return $attachment_id;
+	}
+
+	$attachment_id = 0;
+
+	$results = $wpdb->get_results(
+		$wpdb->prepare(
+			"SELECT post_id, meta_value FROM $wpdb->postmeta WHERE meta_key = '_wp_attached_file' AND meta_value = %s",
+			$path
+		)
+	);
+
+	if ( $results ) {
+		// Use the first available result, but prefer a case-sensitive match, if exists.
+		$attachment_id = reset( $results )->post_id;
```

Comment:
Is the reset technically needed, or can we use `$results[0]`?

Reply:
It is not; just a copy-pasta from the core version. Updated.